data_source.go

```diff
@@ -259,12 +259,12 @@ FROM
 
 	pgCurrentWalLsn, err := ds.getPgCurrentWalLsn(ds.cfg.MaxHop, db)
 	if err != nil {
-		log.Fatalln("Error getting pg_current_wal_lsn:", err)
+		log.Println("Error getting pg_current_wal_lsn:", err)
```

Comment:
If we experience an error in these sections, should we return from the function still?  I'm wondering about potential panics further down (ie. a reference to `pgCurrentWalLsn` where it happens to be nil).

Reply:
Yup. We should probably print but add `return nil, err`.

Reply:
https://github.com/film42/pgreba/blob/2d1cd67c202be1dfa97eaf906899157d4a26cd14/main.go#L37-L41

great point, yes I think we should be returning when an err occurs. If the err is returned, a status 500 is returned

Reply:
That doesn't look like it's returning a 500 more so as a process panic.

I think we should be doing something like...
```
if err != nil {
    http.Error(w, "Some message here", http.StatusInternalServerError)
    return
}
```

Reply:
fixed!